### PR TITLE
test: use Packagist package `ddev/ddev-test-composer-create`, fixes #6586

### LINKED ISSUE #6586: TestComposerCreateCmd doesn't work with Composer 2.8.1
### Actual Behavior

Panic SEGV in pkg/composer/composer.go:65

I note this panic in TestComposerCreateCmd, https://buildkite.com/ddev/macos-colima-vz/builds/1710#01925778-6aa1-439e-a5a2-800234aee613/15326-15327

```


panic: runtime error: invalid memory address or nil pointer dereference [recovered]
  | panic: runtime error: invalid memory address or nil pointer dereference
  | [signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x105300470]
  |  
  | goroutine 737 [running]:
  | testing.tRunner.func1.2({0x1055ce140, 0x105bc7190})
  | /opt/homebrew/Cellar/go/1.23.2/libexec/src/testing/testing.go:1632 +0x1bc
  | testing.tRunner.func1()
  | /opt/homebrew/Cellar/go/1.23.2/libexec/src/testing/testing.go:1635 +0x334
  | panic({0x1055ce140?, 0x105bc7190?})
  | /opt/homebrew/Cellar/go/1.23.2/libexec/src/runtime/panic.go:785 +0x124
  | github.com/ddev/ddev/pkg/composer.(*Manifest).keyExists(...)
  | /Users/testbot/tmp/buildkite-agent/builds/tb-macos-arm64-6/ddev/macos-colima-vz/pkg/composer/composer.go:65
  | github.com/ddev/ddev/pkg/composer.(*Manifest).HasPostRootPackageInstallScript(...)
  | /Users/testbot/tmp/buildkite-agent/builds/tb-macos-arm64-6/ddev/macos-colima-vz/pkg/composer/composer.go:104
  | github.com/ddev/ddev/cmd/ddev/cmd.TestComposerCreateCmd(0x14000334ea0)
  | /Users/testbot/tmp/buildkite-agent/builds/tb-macos-arm64-6/ddev/macos-colima-vz/cmd/ddev/cmd/composer-create_test.go:222 +0x2280
  | testing.tRunner(0x14000334ea0, 0x1056a0cb0)
  | /opt/homebrew/Cellar/go/1.23.2/libexec/src/testing/testing.go:1690 +0xe4
  | created by testing.(*T).Run in goroutine 1
  | /opt/homebrew/Cellar/go/1.23.2/libexec/src/testing/testing.go:1743 +0x314
 ```

Same panic in https://buildkite.com/ddev/macos-lima/builds/1540#01925778-5d35-4f2c-858c-8153bd8e65e8

This may have been going on for some time.

### Steps To Reproduce

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by stasadev**
## The Issue

- #6586

## How This PR Solves The Issue

Uses https://packagist.org/packages/ddev/ddev-test-composer-create instead of a local one.
The functionality that was tested with `--repository` flag is replaced with `--prefer-install` flag.

## Manual Testing Instructions

It should simply pass the tests.

### CODE REVIEW

File: docs/content/developers/maintainers.md [hunk 1]
```diff
@@ -48,6 +48,8 @@ There are cases like access to hosting provider integrations that have essential
 * **CircleCI**: Maintainers automatically have some access via their GitHub team membership, but should probably get more.
 * **developer.apple.com**: Add to the DDEV team there so certificates can be managed.
 * **hub.docker.com**: Add user to owners team in DDEV org.
+* **AUR**: Add user to Co-maintainers (account should be created first) for `ddev-bin` and `ddev-edge-bin`.
+* **Packagist**: Add user to Maintainers (login with GitHub) for `ddev/ddev-test-composer-create`.
```
Comment (stasadev) [inline docs/content/developers/maintainers.md:52]: @rfay, please create an account on https://packagist.org/, so I can add you.

Comment (rfay) [reply docs/content/developers/maintainers.md:52]: Account created, `rfay`, thanks.

Comment (stasadev) [reply docs/content/developers/maintainers.md:52]: Done https://packagist.org/packages/ddev/ddev-test-composer-create

**Review by rfay — APPROVED** [review]
Thanks!

**Pull request merged**